storage/watchable_store.go

```diff
@@ -203,9 +208,9 @@ func (s *watchableStore) watch(key []byte, prefix bool, startRev int64, ch chan<
 			return
 		}
 
-		for i, w := range s.synced[k] {
-			if w == wa {
-				s.synced[k] = append(s.synced[k][:i], s.synced[k][i+1:]...)
+		if v, ok := s.synced[k]; ok {
```

Comment:
if there is nothing in s.synced[k], we should also remove s.synced[k] (empty map item)